### PR TITLE
feat: Bacis catalog setup

### PR DESCRIPTION
**Pull request by connoratrug**
Basic server side rendered (ssr) catalogue 

- Uses nuxt for ssr 
- Moves components needed for ssr ( check for use of browser libs and webpack magic) to molgenis-components lib ( remove form styleguide-dist when done)
- Add build docker image to gradle build
- Instruct circle ci to deploy ssr-catalogue to docker repo on successful ci build and test

### CODE REVIEW

File: apps/nuxt-ssr/pages/_schema/ssr-catalogue.vue [hunk 1]
```diff
@@ -2,8 +2,21 @@
 <div>
   <Menu :menu="menu"/>
   <div class="container pt-3">
-    <h1>Data Catalogue for schema: {{$route.params.schema}}</h1>
-        <h2>Collected data</h2>
+    <h1>
```
Comment (mswertz) [inline apps/nuxt-ssr/pages/_schema/ssr-catalogue.vue:5]: This one we don't really need in ssr? Or is your idea to move whole catalogue into ssr?

Comment (connoratrug) [reply apps/nuxt-ssr/pages/_schema/ssr-catalogue.vue:5]: yes , move the whole catalogue

**Review by mswertz — CHANGES_REQUESTED** [review]
Unclear what this pull request is supposed to do.
Code looks nice

But for examples when I go to preview/apps/components I don't see anything. I would expect to see something??

**Comment by connoratrug** [comment]
> But for examples when I go to preview/apps/components I don't see anything. I would expect to see something??

No these is no components app

> Unclear what this pull request is supposed to do. Code looks nice

Setup the build steps needed to build a docker container serves the ssr catalogue app

File: apps/nuxt-ssr/nuxt.config.js [hunk 2]
```diff
@@ -5,22 +5,22 @@ export default {
...
       { rel: "icon", type: "image/x-icon", href: "/favicon.ico" },
-      { rel: "stylesheet", href: "theme.css" }
-    ]
+      { rel: "stylesheet", href: "theme.css" },
+    ],
   },
 
   // Global CSS: https://go.nuxtjs.dev/config-css
-  css: [],
+  css: ["assets/css/style.css"],
```
Comment (mswertz) [inline apps/nuxt-ssr/nuxt.config.js:23]: how does this interfer with themes?

**Review by mswertz — APPROVED** [review]
Looks good. But why the custom styling? 
I would prefer to not use css and only bootstrap classes unless not possible.
(Or did I do that myself earlier?)

**Comment by connoratrug** [comment]
> Looks good. But why the custom styling? I would prefer to not use css and only bootstrap classes unless not possible. (Or did I do that myself earlier?)

What do you mean by 'custom styling' , it is the same type of styling used in the styleguide dist module. Bootstap only 'bootstraps' the styles, additional styling is needed for some of the views and components we have. 

Not sure if you mean the styling in de nuxt app or the style.css generated by the components lib , both are pretty minimal and the same / comparable the non srr styles.

File: apps/nuxt-ssr/nuxt.config.js (hunk 2 above)
Comment (connoratrug) [reply apps/nuxt-ssr/nuxt.config.js:23]: these are the components styles, like all css if global css is used these would interfer / interact with other ccs ( the c in  css). Most / all components use scoped styles however ( the vue compiler adds data tags to the html to make sure the css only effects the 'scoped' components and does not interfere with other ( theme) css. The effects are the same as the non-ssr styles

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,18 @@
 FROM node:lts-gallium
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply Dockerfile:6]: added comments

File: apps/molgenis-components/src/App.vue [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      return this.$docNames;
+    },
+  },
+};
+</script>
+
+
+
+<style>
+#sidebar-wrapper .sidebar-heading {
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/App.vue:32]: it is using bootstrap , just a bit of added styling ( dev mode only) to have a side menu bar ( not part of the bootstrap component set as far as i know)

File: apps/molgenis-components/src/SideBar.vue [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <div id="sidebar-wrapper" class="border-right">
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/SideBar.vue:2]: yes, still a todo , not too complex but just started with a static list to figure out what a wantend. you use the component files parser in the src/main.js to construct list

File: apps/molgenis-components/src/components/display/ContactDisplay.vue [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        :src="imgSrc"
+        :alt="contact.surname + '-image'"
+        class="img-thumbnail"
+      />
+    </template>
+  </person-details>
+</template>
+
+<style scoped>
+.card {
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/components/display/ContactDisplay.vue:36]: to fit in with the rest of the molgenis  / catalogue styling

File: apps/molgenis-components/src/components/display/ContactDisplay.vue [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+  padding: 0;
+  border-radius: 0;
+}
+</style>
+
+<script>
+import PersonDetails from "./PersonDetails.vue";
+
+export default {
+  name: "ContactDisplay",
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/components/display/ContactDisplay.vue:50]: yes adding a name is part of the 'best practices' described by the vue team in the vue style guide, see https://v2.vuejs.org/v2/api/?redirect=true#name for details

File: apps/molgenis-components/src/components/tables/cellTypes/ListDisplay.vue [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+          focusable="false"
+          data-prefix="fas"
+          data-icon="chevron-down"
+          class="svg-inline--fa fa-chevron-down fa-w-14"
+          role="img"
+          xmlns="http://www.w3.org/2000/svg"
+          viewBox="0 0 448 512"
+        >
+          <path
+            fill="currentColor"
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/components/tables/cellTypes/ListDisplay.vue:21]: chevron-down svg image

File: apps/molgenis-components/src/docs/display/ImageDisplay.vue [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
...QGrlLLzxOroobXIiqCH5aMBAtaqQydjdQLxTGYr6siTkxC0InUBCHhHC9wMnyEY0XzL91NH9lunIejKNhzqm/IYGr4WqcJMQHs10vGpI2+rJSCDobT9KEVA22ze4kMEIvaBv9ceaiToItn9oY7cr5aAjnii8U2agPbd+GcDCaiQsc6HHm7DREB7IWIQwbWtWoLYMgS8BbzOIxCDHOPxOo1gIR/KHHlddRnE45sMAe1LsS5sDgH1BTbYIPpGK41gJF2ZPTyrqYf6QXchLeoX9JLPxYQ0j6A1YaNMwzFTjaDlifiz1JFnhxGMesIW8e9DbRHPMO325xKII6JEmx7SCabiX13C8nEIwURuFSzsVYhfon/IJxCxTPIiMEkCVicXrETZsw8iEMOXwp6doGUdhyICHjrAz8y36AQL7lXCkQ8uA2RJ0OKDhEICFjOQZNpUnYC58hN7NvuPQwhWw+E4PULKIWD9saVXRMDjyKbD2RkIeFFRWeAac4rxAf8THSSQp5mRgG1ajTLNCgGn57wMBPzfwpFPRyAjV5kItlQ/4onamdZNNRBwV5YJT0eQmSlRCAb0jzEBVW84pmBEwEeasm8B5IARyIGQaa4iafHWuGiuIiJYYGuCwjLgYyIvplHmi5JJlGQSSiNIJgdEFiVOEUELJOgiZbp3pnm4iWDA6yTvYycEPXXmbJi0exrBGitzTvt4pqSQYIoAgg5KErHBDk4LgomAV8lYleRD+exlRO/Ftw8Y5k2x4mwonrwuJODzgNCsYzRXMVQ/D0zAI6hLrUyqBJ9BFvgstlE0e8AI0hWThfSTxc4Syrm9QgJ+EQY00pdVkVeETLfCRMC7A6LVVQh4bC+/Oxz3EVKC+7H7l2eREe47POFOJoxaDj4+iBPOvGgFJCFgTzISsBEaIp3xkJ/xzk68mAi4U8Uw8UP5OIkgfmQoicfDb5Am0sQvqQnjNSyxHhSbrxOwojdrMg+9gsQpHy2mnZGAr920sgStfnygFCfBVLqGsVeckOBtRGpchVqqBMzTzASjWXQqmvhaTVQIpr4vc9ChfTJRSyZsRTJOsdjxDrPvKa67Y2uM/H/s/zK/i4HPUvqzxEP6URKZmnvUkv4pSTJQXjajSRT3GnvizTv9AFjJ9eTRerlc2JxkGi1owirP3qzoA9emJeX/A0azrvJNMdx4AAAAAElFTkSuQmCC"
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/docs/display/ImageDisplay.vue:9]: yes , using inline svg avoids the need to reference external files and avoid path issues

File: apps/molgenis-components/src/components/tables/cellTypes/ListDisplay.vue (hunk 8 above)
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/tables/cellTypes/ListDisplay.vue:21]: aaah, I use fontawesome. Why don't you? Are you one of these guys that prefers to embed icons?

**Comment by mswertz** [comment]
> What do you mean by 'custom styling' , it is the same type of styling used in the styleguide dist module. Bootstap only 'bootstraps' the styles, additional styling is needed for some of the views and components we have.

I meant: I noticed quite some style blocks, where I though we could do this using bootstrap classes like m-0, p-0 etc.

File: apps/molgenis-components/src/components/tables/cellTypes/ListDisplay.vue (hunk 8 above)
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/components/tables/cellTypes/ListDisplay.vue:21]: i am using font awesome , the path was drawn by fontawesome , but not loading saves a dependency. imo svg support removes the need to include these types of libs. We could include it , but that just means more setup and magic imo , once the build step generated the build file the same path is going to be included.

File: apps/nuxt-ssr/assets/css/style.css [hunk 10]
```diff
@@ -0,0 +1 @@
+.card[data-v-7e087a58]{border:0}.img-thumbnail[data-v-7e087a58]{border:0;padding:0;border-radius:0}.card[data-v-2ec27f48]{border:0}.img-thumbnail[data-v-2ec27f48]{border:0;padding:0;border-radius:0}.block[data-v-744392e4]{margin-bottom:1rem;padding:1rem}.image-card>img{width:100%;border-top-left-radius:-1px;border-top-right-radius:-1px}.card-group,.card-columns>.image-card>img{border-radius:0/0px!important}ul.list-group li.list-group-item[data-v-6d3af544]{border:none}ul.list-group li.list-group-item[data-v-6d3af544]{padding-left:0}ul.list-group.list-group-horizontal li.list-group-item[data-v-6d3af544]{padding-top:0;padding-bottom:0}img[data-v-3693b77a]{height:2.5rem}.table-hover tbody tr:hover{cursor:pointer}
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/nuxt-ssr/assets/css/style.css:1]: scoped styles from the component lib

**Pull request merged**